admin-dev/themes/new-theme/js/pages/product/price-refresher.js

```diff
@@ -0,0 +1,153 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import productMap from './product-map';
+import * as types from './store/mutation-types';
+
+/**
+ * Responsible for price tax excluded/included and tax rate selection inputs interaction
+ */
+export default class PriceRefresher {
+  constructor(store) {
+    this.store = store;
+    this.taxIncludedInputs = productMap.priceTaxIncludedInputs;
+    this.taxExcludedInputs = productMap.priceTaxExcludedInputs;
+    this.taxRulesGroupSelections = productMap.taxRulesGroupSelections;
+
+    this.init();
+  }
+
+  /**
+   * Initializes two way binding between DOM and state.
+   */
+  init() {
+    this.listenToDomChanges();
+    this.listenToStateChanges();
+  }
+
+  /**
+   * Listens to DOM elements changes and mutates the state accordingly.
+   */
+  listenToDomChanges() {
+    this.listenToPriceChangesInDom(this.taxIncludedInputs, true);
+    this.listenToPriceChangesInDom(this.taxExcludedInputs, false);
+    this.listenToTaxRulesGroupChangesInDom();
+  }
+
+  /**
+   * Subscribes to state mutations and updates the DOM.
+   */
+  listenToStateChanges() {
+    this.store.subscribe((mutation) => {
+      const subscribedMutationTypes = [
+        types.SET_PRICE_TAX_INCLUDED,
+        types.SET_PRICE_TAX_EXCLUDED,
+        types.SET_TAX_RULES_GROUP,
+      ];
+
+      if (!subscribedMutationTypes.includes(mutation.type)) {
+        return;
+      }
+
+      if (mutation.type === types.SET_PRICE_TAX_INCLUDED) {
+        this.updatePricesInDom(this.taxIncludedInputs, true);
+      } else if (mutation.type === types.SET_PRICE_TAX_EXCLUDED) {
+        this.updatePricesInDom(this.taxExcludedInputs, false);
+      } else {
+        this.updateTaxRulesGroupInDom();
+      }
+    });
+  }
+
+  /**
+   * Listens for price tax included/excluded inputs changes
+   *
+   * @param selector
+   * @param forTaxIncluded
+   */
+  listenToPriceChangesInDom(selector, forTaxIncluded) {
+    const elements = document.querySelectorAll(selector);
+
+    elements.forEach((el) => {
+      el.addEventListener('change', (event) => {
+        if (forTaxIncluded) {
+          this.store.dispatch('updatePriceTaxIncluded', {
+            priceTaxIncluded: event.currentTarget.value,
+          });
+        } else {
+          this.store.dispatch('updatePriceTaxExcluded', {
+            priceTaxExcluded: event.currentTarget.value,
+          });
+        }
+      });
+    });
+  }
+
+  /**
+   * Listens to tax rules selection changes
+   */
+  listenToTaxRulesGroupChangesInDom() {
+    const elements = document.querySelectorAll(this.taxRulesGroupSelections);
```

Comment:
In this case the store binder could work as well as long as the store contains the list of `taxRulesGroups` The we just need to have a `updateTaxRulesGroupId` that will match the rate thanks to the id